src/pytorch_lightning/strategies/launchers/spawn.py

```diff
@@ -69,13 +78,19 @@ def launch(self, function: Callable, *args: Any, trainer: Optional["pl.Trainer"]
                 a selected set of attributes get restored in the main process after processes join.
             **kwargs: Optional keyword arguments to be passed to the given function.
         """
+        if self._start_method not in mp.get_all_start_methods():
```

Comment:
Yes, in general we can't just fork a process on the cpu if we already have data in cuda, like tensors etc. Which is why they say it's not supported. Hence also the note in the docs that you shouldn't call any cuda related functions before calling trainer.fit(). However, in Lightning we can control all of this and ensure we create processes before we setup the model on GPU or move any data there. 

This is the reason why https://github.com/Lightning-AI/lightning/issues/7550 was open for so long, because until recently Lightning didn't spawn processes early enough for this to become possible.